mat64/list_test.go

```diff
@@ -373,6 +392,114 @@ func underlyingData(a Matrix) []float64 {
 	}
 }
 
+// testMatrixTypes is a list of matrix types to test.
+var testMatrices = []Matrix{
+	&Dense{},
+	&SymDense{},
+	NewTriDense(0, true, nil),
+	NewTriDense(0, false, nil),
+	NewVector(0, nil),
+	&Vector{mat: blas64.Vector{Inc: 10}},
+	&basicMatrix{},
+	&basicVectorer{},
+	&basicSymmetric{},
+	&basicTriangular{},
+
+	Transpose{&Dense{}},
+	Transpose{NewTriDense(0, true, nil)},
+	TransposeTri{NewTriDense(0, true, nil)},
+	Transpose{NewTriDense(0, false, nil)},
+	TransposeTri{NewTriDense(0, false, nil)},
+	Transpose{NewVector(0, nil)},
+	Transpose{&Vector{mat: blas64.Vector{Inc: 10}}},
+	Transpose{&basicMatrix{}},
+	Transpose{&basicVectorer{}},
+	Transpose{&basicSymmetric{}},
+	Transpose{&basicTriangular{}},
+}
+
+func testOneInputFunc(c *check.C,
+	// name is the name of the function being tested.
+	name string,
+
+	// f is the function being tested.
+	f func(a Matrix) interface{},
+
+	// denseComparison performs the same operation, but using Dense matrices for
+	// comparison.
+	denseComparison func(a *Dense) interface{},
+
+	// sameAnswer compares the result from two different evaluations of the function
+	// and returns true if they are the same. The specific function being tested
+	// determines the definition of "same". It may mean identical or it main mean
+	// approximately equal.
+	sameAnswer func(a, b interface{}) bool,
+
+	// legalType returns true if the type of the input is a legal type for the
+	// input of the function.
+	legalType func(a Matrix) bool,
+
+	// legalSize returns true if the size is valid for the function.
+	legalSize func(r, c int) bool,
+) {
+	for _, aMat := range testMatrices {
+		for _, test := range []struct {
+			ar, ac int
+		}{
+			{1, 1},
+			{1, 3},
+			{3, 1},
+
+			{6, 6},
+			{6, 11},
+			{11, 6},
+		} {
+			// Skip the test if any argument would not be assignable to the
```

Comment:
I changed it to "the" because there is only one argument.

In general though, they aren't the same. "any argument would not be assignable" = "at least one argument would not be assignable". "no argument would be assignable" = "all arguments are not assignable".